File: Java/Bicchiere/src/me/gca/Bicchiere.java
package me.gca;

public class Bicchiere {

    private String materiale;
    private String forma;
    private float livello;
    private float capacita;

    public Bicchiere(){}

    public Bicchiere(String materiale, String forma, float livello, float capacita) {
        this.materiale = materiale;
        this.forma = forma;
        this.livello = livello;
        this.capacita = capacita;
    }

    public String getMateriale() {
        return materiale;
    }

    public String getForma() {
        return forma;
    }

    public float getLivello() {
        return livello;
    }

    public float getCapacita() {
        return capacita;
    }

    public void setMateriale(String materiale) {
        this.materiale = materiale;
    }

    public void setForma(String forma) {
        this.forma = forma;
    }

    public void setLivello(float livello) {
        this.livello = livello;
    }

    public void setCapacita(float capacita) {
        this.capacita = capacita;
    }

    public void riempi(){
        livello = capacita;
    }

    public void svuota(){
        livello = 0;
    }
}

File: C++/Bicchiere_Esempio/bicchiere.cpp
#include "bicchiere.h"


/* metodi costruttori */
Bicchiere::Bicchiere()
{
   strcpy(forma,"");
   strcpy(materiale,"");
   capacita = 0;
   livello = 0;
   printf("nuovo bicchiere creato!\n");
}

Bicchiere::Bicchiere(char *f, char *m, float c, float l)
{
   strcpy(forma, f);
   strcpy(materiale, m);
   capacita = c;
   livello = l;
   printf("nuovo bicchiere creato!\n");
}

/* metodo distruttore */
Bicchiere::~Bicchiere()
{
   printf("oggetto eliminato!\n");                                         
}

/* metodi get */
char *Bicchiere::GetForma()
{
     return forma;     
}

char *Bicchiere::GetMateriale()
{
     return materiale;     
}

float Bicchiere::GetCapacita()
{
      return capacita;      
}
     
float Bicchiere::GetLivello()
{
      return livello;      
}

/* metodi set */
void Bicchiere::SetForma(char *f)
{
     strcpy(forma, f);     
}

void Bicchiere::SetMateriale(char *m)
{
     strcpy(materiale, m);     
}

void Bicchiere::SetCapacita(float c)
{
     capacita = c;     
} 

void Bicchiere::SetLivello(float l)
{
     livello = l;     
}
     
void Bicchiere::Riempi()
{
     livello = capacita;     
}

void Bicchiere::Svuota()
{
     livello = 0;     
}
